app/presenters/tree_node/menu/node.rb

```diff
@@ -0,0 +1,39 @@
+module TreeNode
+  module Menu
+    class Node < TreeNode::Node
+      set_attribute(:no_click, true)
+      set_attribute(:icon, "pficon pficon-folder-close")
+
+      set_attribute(:checkable) { @options[:editable] }
+
+      def self_and_child_states
+        child_states.unshift(selected)
+      end
+
+      private
+
+      def child_states
+        @states ||= children.flat_map(&:self_and_child_states)
+      end
+
+      def select_by_children_selected
+        return true        if all_children_fully_selected?
+        return 'undefined' if any_children_selected?
+
```

Comment:
Could you please move [these](https://github.com/ManageIQ/manageiq-ui-classic/pull/137/files#diff-132f31a85f7a310876d397d8d3a991fdR19) [three](https://github.com/ManageIQ/manageiq-ui-classic/pull/137/files#diff-132f31a85f7a310876d397d8d3a991fdR26) [methods](https://github.com/ManageIQ/manageiq-ui-classic/pull/137/files#diff-132f31a85f7a310876d397d8d3a991fdR32) into a single one and add some comments if necessary?